src/tools/unicode-table-generator/src/case_mapping.rs

```diff
@@ -47,8 +46,43 @@ fn generate_tables(case: &str, data: &BTreeMap<u32, [u32; 3]>) -> (String, usize
         mappings.push((CharEscape(key), value));
     }
 
-    let mut tables = String::new();
     let mut size = 0;
+    let mut tables = String::new();
+    writeln!(
+        tables,
+        "\
+#[inline]
+pub fn to_{case_lower}(c: char) -> [char; 3] {{
+    const {{
+        let mut i = 0;
+        while i < {case_upper}CASE_TABLE.len() {{
+            let (_, val) = {case_upper}CASE_TABLE[i];
+            if val & (1 << 22) == 0 {{
+                assert!(char::from_u32(val).is_some());
+            }} else {{
+                let index = val & ((1 << 22) - 1);
+                assert!((index as usize) < {case_upper}CASE_TABLE_MULTI.len());
+            }}
+            i += 1;
+        }}
+    }}
+
+    // SAFETY: Just checked that the tables are valid
+    unsafe {{
+        super::case_conversion(
+            c,
+            |c| c.to_ascii_{case_lower}case(),
+            {case_upper}CASE_TABLE,
+            {case_upper}CASE_TABLE_MULTI,
+        )
+    }}
```

Comment:
Why are there non-trivial changes mixed in with a commit that says it's only moving functions from one place to another?